### PR TITLE
Add `GenericWindow`

### PR DESCRIPTION
**Pull request by saihaze**
`GenericWindow` is added to abstract over both Wayland and X11 windows.
Some comments are copied from `Window`.
Anvil hasn't been modified yet. It's WIP.

### CODE REVIEW

**Comment by saihaze** [comment]
Related to #1202

File: anvil/src/shell/element.rs [hunk 1]
```diff
@@ -88,10 +79,10 @@ impl WindowElement {
         T: Into<Duration>,
         F: FnMut(&WlSurface, &WlSurfaceData) -> Option<Output> + Copy,
     {
-        match self {
-            WindowElement::Wayland(w) => w.send_frame(output, time, throttle, primary_scan_out_output),
+        match &self.0 {
+            GenericWindow::Wayland(w) => w.send_frame(output, time, throttle, primary_scan_out_output),
             #[cfg(feature = "xwayland")]
-            WindowElement::X11(w) => {
+            GenericWindow::X11(w) => {
```
Comment (Drakulix) [inline anvil/src/shell/element.rs:84]: I think this piece of code is a good place to create a `GenericWindow::send_frames` method instead.

**Review by Drakulix — COMMENTED** [review]
Looks quite useful already!

File: anvil/src/shell/element.rs (hunk 1 above)
Comment (saihaze) [reply anvil/src/shell/element.rs:84]: OK. Let me do it today.

Comment (saihaze) [reply anvil/src/shell/element.rs]: Done.

**Pull request closed without merging**